### ISSUE TITLE
hexagon-unknown-linux-musl: libstandalone incorrectly included

### ISSUE THREAD

**Issue by androm3da** [llvm/llvm-project#64518]
A bug was introduced in 96832a6bf7e0e7f1e8d634d38c44a1b32d512923 that caused the driver to include `libstandalone` and other unintended content for linux program link.

```
$ ./bin/clang ./bin/clang --target=hexagon-unknown-linux-musl  -o tprog tprog.o -L/tmp -###
...
... "/usr/lib/crt1.o" "-L/usr/lib" "./bin/clang" "tprog.o" "-lclang_rt.builtins-hexagon" "-lc" "/local/mnt/workspace/upstream/obj_ubuntu/./bin/hexagon/lib/v60/crt0_standalone.o" "/local/mnt/workspace/upstream/obj_ubuntu/./bin/hexagon/lib/v60/crt0.o" "/local/mnt/workspace/upstream/obj_ubuntu/./bin/hexagon/lib/v60/init.o" "-L/tmp" "-L/local/mnt/workspace/upstream/obj_ubuntu/./bin/hexagon/lib/v60" "-L/local/mnt/workspace/upstream/obj_ubuntu/./bin/hexagon/lib" "./bin/clang" "tprog.o" "--start-group" "-lstandalone" "-lc" "-lgcc" "--end-group" "/local/mnt/workspace/upstream/obj_ubuntu/./bin/hexagon/lib/v60/fini.o"

```

**Comment by androm3da** [comment]
@kparzysz-quic can you approve cherry-picking the fix for this bug in 5bc4b34a3aa9c6ea10663a252ac46d20862b38d5 to release/17.x?

**Comment by kparzysz-quic** [comment]
Approved
